service/qos/service_level_controller.cc

```diff
@@ -687,6 +689,71 @@ future<> service_level_controller::unregister_subscriber(qos_configuration_chang
     return _subscribers.remove(subscriber);
 }
 
+static sstring describe_service_level(std::string_view sl_name, const service_level_options& sl_opts) {
+    using slo = service_level_options;
+
+    utils::small_vector<sstring, 2> opts{};
+    
+    const sstring sl_name_formatted = cql3::util::maybe_quote(sl_name);
+
+    if (auto maybe_timeout = std::get_if<lowres_clock::duration>(&sl_opts.timeout)) {
+        // According to the documentation, `TIMEOUT` has to be expressed in milliseconds
+        // or seconds. It is therefore safe to use milliseconds here.
+        const auto timeout = std::chrono::duration_cast<std::chrono::milliseconds>(*maybe_timeout);
+        opts.push_back(seastar::format("TIMEOUT = {}", timeout));
+    }
+
+    switch (sl_opts.workload) {
+        case slo::workload_type::batch:
+            opts.push_back("WORKLOAD_TYPE = 'batch'");
+            break;
+        case slo::workload_type::interactive:
+            opts.push_back("WORKLOAD_TYPE = 'interactive'");
+            break;
+        case slo::workload_type::unspecified:
+            break;
+        case slo::workload_type::delete_marker:
+            on_internal_error(sl_logger, "Precondition has been violated");
+    }
+
+    switch (opts.size()) {
+        case 0:
+            return seastar::format("CREATE SERVICE_LEVEL {};", sl_name_formatted);
+        case 1:
+            return seastar::format("CREATE SERVICE_LEVEL {} WITH {};", sl_name_formatted, opts[0]);
+        case 2:
+            return seastar::format("CREATE SERVICE_LEVEL {} WITH {} AND {};", sl_name_formatted, opts[0], opts[1]);
```

Comment:
You can write: `...WITH {}", sl_name_formatted, opts | std::views::join_with(sstring(" AND ")));` and gain some option count independence.